x/upgrade/spec/01_concepts.md

```diff
@@ -0,0 +1,73 @@
+# Concepts
+
+## Plan
+
+The `x/upgrade` module defines a `Plan` type in which a live upgrade is scheduled
+to occur. A `Plan` can be scheduled at a specific block height or time, but not both.
+A `Plan` is created once a (frozen) release candidate along with an appropriate upgrade
+`Handler` (see below) is agreed upon, where the `Name` of a `Plan` corresponds to a
+specific `Handler`. Typically, a `Plan` is created through a governance proposal
+process, where if voted upon and passed, will be scheduled. The `Info` of a `Plan`
+may contain various metadata about the upgrade, typically application specific
+upgrade info to be included on-chain such as a git commit that validators could
+automatically upgrade to.
```

Comment:
Not strictly necessary, but we may want to link to the cosmosd format for `Info` described here: https://github.com/regen-network/cosmosd#auto-download. This format will allow cosmosd to automatically download and checksum binaries.